qemu/tests/cpu_topology_test.py

```diff
@@ -0,0 +1,50 @@
+import random
+
+from virttest import error_context
+from virttest import env_process
+
+from provider.cpu_utils import check_guest_cpu_topology
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def check(p_name, exp, check_cmd):
+        """
+        Check the cpu property inside guest
+        :param p_name: Property name
+        :param exp: The expect value
+        :param check_cmd: The param to get check command
+        """
+        res = session.cmd_output(params[check_cmd]).strip()
+        if int(res) != int(exp):
+            test.fail('The vcpu %s number inside guest is %s,'
+                      ' while it is set to %s' % (p_name, res, exp))
+
+    os_type = params['os_type']
+    params['vcpu_cores'] = vcpu_cores = random.randint(1, 10)
+    vcpu_threads_list = [1, 2]
+    if params['machine_type'] == 'pseries':
+        vcpu_threads_list = [1, 2, 4, 8]
+    vcpu_sockets = random.randint(1, 10)
+    params['vcpu_sockets'] = 2 if (os_type == 'Windows' and
+                                   vcpu_sockets > 2) else vcpu_sockets
+    vm_name = params['main_vm']
+    for vcpu_threads in vcpu_threads_list:
+        params['vcpu_threads'] = vcpu_threads
+        params['smp'] = params['vcpu_maxcpus'] = (vcpu_cores *
+                                                  vcpu_threads * vcpu_sockets)
```

Comment:
On ppc, the most extreme case is `smp = 10 * 8 * 10`, which is larger than vCPU limitation(240/384), please limit vcpu_cores and vcpu_sockets.

Reply:
@PaulYuuu Updated to `vcpu_sockets = 240 // (vcpu_cores * vcpu_threads)`.